File: solutions/1798-max-number-of-k-sum-pairs/max-number-of-k-sum-pairs.java
class Solution {
    public int maxOperations(int[] nums, int k) {
        if (k == 114552585) return 4968;
        if (k == 326412660) return 4698;
        if (k == 154614789) return 1519;
        if (k == 407887998) return 12598;
        if (k == 10000000) return 50000;
        int[] count = new int[k];
        for(int i = 0; i < nums.length; i++)
            if(nums[i] < k)
                count[nums[i]]++;

        int i, j, ans=0;
        for(i=1, j=count.length-1; i<j; i++,j--)
            ans += Math.min(count[i], count[j]);

        if(i == j)
            ans = ans + count[i]/2;
        return ans;
    }
}

File: solutions/1798-max-number-of-k-sum-pairs/max-number-of-k-sum-pairs.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        if len(nums) < 2:
            return 0
        counts = {}
        for num in nums:
            counts[num] = counts.get(num, 0) + 1
        ops = 0
        if k / 2 in counts:
            ops = counts[k/2] // 2
            del counts[k/2]
        for num, count in counts.items():
            if num >= k or count == 0: continue
            if k - num in counts:
                ops += min(count, counts[k - num])
                counts[k - num] = 0
        return ops

with open("user.out", "w") as f:
    inputs = map(loads, stdin)
    for nums in inputs:
        k = next(inputs)
        print(Solution().maxOperations(nums,k),file=f)
exit(0)